src/main/java/com/jcabi/jdbc/StoredProcedureOutcome.java

```diff
@@ -0,0 +1,110 @@
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.jdbc;
+
+import com.jcabi.aspects.Immutable;
+import java.sql.CallableStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+/**
+ * Outcome of a stored procedure with OUT parameters.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.17
+ * @param <T> Type of items
+ */
+@Immutable
+@ToString
+@EqualsAndHashCode(of = "indexes")
+public final class StoredProcedureOutcome<T> implements Outcome<T> {
+
+    /**
+     * OUT parameters' indexes.
+     */
+    @Immutable.Array
+    private final transient int[] indexes;
+
+    /**
+     * Ctor.
+     * @param params Number of OUT params. Has to be > 0.
+     *  If this ctor is used, it is assumed that the OUT parameters
+     *  are from index 1 to including nrop.
+     */
+    public StoredProcedureOutcome(final int params) {
+        if (params <= 0) {
+            throw new IllegalArgumentException(
+                "Nr of out params has to be a positive int!"
+            );
+        }
+        this.indexes = new int[params];
+        for (int idx = 0; idx < params; ++idx) {
+            this.indexes[idx] = idx + 1;
+        }
+    }
+
+    /**
+     * Ctor.
+     * @param indexes Indexes of the OUT params.
+     *  <b>Index count starts from 1</b>.
+     */
+    public StoredProcedureOutcome(final int... indexes) {
+        if (indexes.length == 0) {
+            throw new IllegalArgumentException(
+                "At least 1 OUT param's index needs to be specified!"
+            );
+        }
+        final int size = indexes.length;
+        this.indexes = new int[size];
+        for (int idx = 0; idx < size; ++idx) {
+            this.indexes[idx] = indexes[idx];
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public T handle(
+        final ResultSet rset, final Statement stmt) throws SQLException {
+        final int params = this.indexes.length;
+        final Object[] outs = new Object[params];
+        if (stmt instanceof CallableStatement) {
+            for (int idx = 0; idx < params; ++idx) {
+                outs[idx] = ((CallableStatement) stmt).getObject(
+                    this.indexes[idx]
+                );
+            }
+        }
+        return (T) outs;
```

Comment:
@mkordas It doesn't get any better than this - I tried every combination of T and ``Object[]`` (actually this casting and generics stuff is good candidate for exam questions, you pretty much need to have a compiler in your head to know everything). Initially, I figured maybe it would simply let me return ``Object``, but it doesn't.